### ISSUE TITLE
API: Add force deletion option

### ISSUE THREAD

**Issue by yasker** [rancher/rancher#17823]
When a node is down, the only way to remove the pod from the node is through force deletion, as described in https://kubernetes.io/docs/tasks/run-application/force-delete-stateful-set-pod/ .

Rancher UI doesn't have the ability to force delete a pod. In Kubernetes API, force deletion is performed by call delete with `--grace-period=0`. We need this option for Pod. In fact, every object can be force deleted, but we can limit the first implementation in Pod.

It can be shown as a checkbox when the user clicks the `delete` button of the Pod, along with warning like: `force deletion will not allow the application to gracefully shutdown, and should be only be used when necessary, e.g. when the node is down`.

Without this action, there is nothing users can do with the pods on a down node.

**Comment by loganhz** [comment]
Requested by @pengjiang80 's customer

**Comment by jiaqiluo** [comment]
The bug is reproduced in Rancher: v2.2.7

Steps:
- add an RKE EC2 cluster with 1 etcd + control plane, 3 workers
- deploy a daemon set named `test-daemonset`
- deploy a workload named `test-deployment` setting pods deployed to the node `worker-3`
- wait for both workloads to be active
- go to AWS console to shut down the instance `worker-3`

Results:
- in Rancher UI, the node `worker-3` shows `Unavailable`

<img width="1409" alt="Screen Shot 2019-08-16 at 2 09 50 PM" src="https://user-images.githubusercontent.com/6218999/63198422-94906e00-c02f-11e9-83ae-98adbfdf736e.png">

- in both workloads `test-daemonset` and `test-deployment`, the state of the pod running in `worker-3` shows `Unkown`

<img width="1461" alt="Screen Shot 2019-08-16 at 1 57 08 PM" src="https://user-images.githubusercontent.com/6218999/63198468-b984e100-c02f-11e9-9a82-c59b9cf3b502.png">
<img width="1427" alt="Screen Shot 2019-08-16 at 1 57 17 PM" src="https://user-images.githubusercontent.com/6218999/63198469-b984e100-c02f-11e9-8eac-1474e5b13934.png">

More info:

- deleting the pod from UI does not work
- run the kubectl command can delete the pod 
```
kubectl delete pods <pod_name> --grace-period=0 --force -n <namespace>
```

**Comment by jiaqiluo** [comment]
The bug fix is validated in Rancher: master-head 5216a35d


- Follow the same steps as my previous comment says to set the environment up
- Go to delete the pod, now Rancher UI has the option `Force Delete` 
- check the option `Force Delete`  and delete the pod 
- the pod is deleted successfully
